Marlin/src/gcode/host/M116.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2019 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../gcode.h"
+#include "../../inc/MarlinConfig.h"
+
+/**
+ * M116: Machine Size string
+ */
+void GcodeSuite::M116() {
+
+  SERIAL_ECHOPAIR("x_min_bed:", X_MIN_BED, " x_max_bed:", X_MAX_BED, " y_min_bed:", Y_MIN_BED, " y_max_bed:", Y_MAX_BED);
+  SERIAL_ECHOPAIR(" x_min_pos:", X_MIN_POS, " x_max_pos:", X_MAX_POS);
+  SERIAL_ECHOPAIR(" y_min_pos:", Y_MIN_POS, " y_max_pos:", Y_MAX_POS);
+  SERIAL_ECHOLNPAIR(" z_min_pos:", Z_MIN_POS, " z_max_pos:", Z_MAX_POS);
+  
+}
```

Comment:
Note that the bed size and bed limits may change on Dual X Carriage setups and other machines with interesting carriage arrangements depending on which tool is currently selected. This information in its most up to date form is reported in the output of `M503`.

On further inspection of the `M360` command, it also duplicates the purpose of `M503`. It isn't required to include all the keys in the output, so `M360` is free to only include things not included in the output of `M115` or `M503`. However, we then run into the issue that `M360` is used in Marlin for a SCARA function.

Seems to me the best place to report the machine build area (including the Z height) would be in the output of `M115` where it also includes the number of extruders. I believe that host developers will not have a problem with this because they only store the item-with-colon values they recognize. Other solutions would require them to sign on to new G-codes like this one, and other firmware developers are slow to catch on.

Reply:
we can include this in `M115`, to me also it will be the best spot if  `M116` is not available as details from `M115` is also used for the same purpose.

Reply:
updated commit to include limits for dual x carriage setup